samples/HttpRequestFeaturesv2/Controllers/PersonalizerController.cs

```diff
@@ -0,0 +1,176 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.CognitiveServices.Personalizer;
+using Microsoft.Azure.CognitiveServices.Personalizer.Featurizers;
+using Microsoft.Azure.CognitiveServices.Personalizer.Models;
+using System;
+using System.Collections.Generic;
+
+namespace HttpRequestFeaturesExample.Controllers
+{
+    [Route("api/[controller]")]
+    public class PersonalizerController : Controller
+    {
+        PersonalizerClient client;
+
+        public PersonalizerController(PersonalizerClient personalizerClient)
+        {
+            this.client = personalizerClient;
+        }
+
+        /// <summary>
+        /// Creates a RankRequest with user time of day, HTTP request features,
+        /// and taste as the context and several different foods as the actions
+        /// </summary>
+        /// <returns>RankRequest with user info</returns>
+        [HttpGet("GenerateRank")]
+        public RankRequest GenerateRank()
+        {
+            string eventId = Guid.NewGuid().ToString();
+
+            // Get the actions list to choose from personalizer with their features.
+            IList<RankableAction> actions = GetActions();
+
+            // Get context information from the user.
+            HttpRequestFeatures httpRequestFeatures = GetHttpRequestFeaturesFromRequest(Request);
+            string timeOfDayFeature = GetUsersTimeOfDay();
+            string tasteFeature = GetUsersTastePreference();
+
+            // Create current context from user specified data.
+            IList<object> currentContext = new List<object>() {
+                    new { time = timeOfDayFeature },
+                    new { taste = tasteFeature },
+                    new { httpRequestFeatures }
+            };
+
+            // Exclude an action for personalizer ranking. This action will be held at its current position.
+            IList<string> excludeActions = new List<string> { "juice" };
+
+            // Rank the actions
+            return new RankRequest(actions, currentContext, excludeActions, eventId);
+        }
+
+        /// <summary>
+        /// Sends a rank to the personalizer client
+        /// </summary>
+        /// <param name="rankRequest"></param>
+        /// <returns>RankResponse from request</returns>
+        [HttpPost("PostRank")]
+        public RankResponse PostRank([FromBody]RankRequest rankRequest)
+        {
+            return client.Rank(rankRequest);
+        }
+
+        /// <summary>
+        /// Creates a RewardRequest with value of 0
+        /// </summary>
+        /// <returns>RewardRequest with value of 0</returns>
+        [HttpGet("GenerateReward")]
+        public RewardRequest GenerateReward()
+        {
+            // Create reward request with value of 0
+            return new RewardRequest(0);
```

Comment:
>0 [](start = 37, length = 1)

Why default to 0 since generally we want a reward of 1. 0 is the default for no reward.